# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
# Commit: d5cc6cfba0ee
# Full Hash: d5cc6cfba0eecc9e67d70f54865cd4f3220cfce4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-01-11 21:56:57
# Regressor Bug: 1806605
# File Overlap Count: 1
# Description:
#   Bug 1806605 - In VideoCaptureAvFoundation track conversions to I420 with a marker. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D165825
# ==============================================================================

diff -r e85b278387fc -r d5cc6cfba0ee dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Wed Jan 11 11:47:45 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Wed Jan 11 11:47:45 2023 +0000
@@ -223,8 +223,9 @@
 
 int32_t VideoCaptureAvFoundation::OnFrame(webrtc::VideoFrame& aFrame) {
   MutexLock lock(&api_lock_);
+  mConversionRecorder.Record(0);
   int32_t rv = DeliverCapturedFrame(aFrame);
-  mPerformanceRecorder.Record(0);
+  mCaptureRecorder.Record(0);
   return rv;
 }
 
@@ -260,10 +261,13 @@
         return CaptureStage::ImageType::Unknown;
     }
   };
-  mPerformanceRecorder.Start(
+  mCaptureRecorder.Start(
       0, "VideoCaptureAVFoundation"_ns, *mTrackingId, aWidth, aHeight,
       mCapability.map([&](const auto& aCap) { return fromWebrtcVideoType(aCap.videoType); })
           .valueOr(CaptureStage::ImageType::Unknown));
+  if (mCapability && mCapability->videoType != webrtc::VideoType::kI420) {
+    mConversionRecorder.Start(0, "VideoCaptureAVFoundation"_ns, *mTrackingId, aWidth, aHeight);
+  }
 }
 
 void VideoCaptureAvFoundation::MaybeRegisterCallbackThread() {
